### PR TITLE
Respect amount on interact_user_followers

### PR DESCRIPTION
**Pull request by texmex44**
At final of interact_user_followers restart if followed < amount. (if param respectAmount is True)

### CODE REVIEW

**Comment by timgrossmann** [comment]
@texmex44 Thank you for the contribution!
This is a good catch, however I'm not sure if we should use recursion here.
Would you mind changing that to e.g. a while loop?

**Comment by sionking** [comment]
Recursion with no stop condition == memory overflow
We need to add another stop conditions that is unconditioned in IG response (e,g a counter).

**Comment by texmex44** [comment]
@timgrossmann While loop is here 👍 Can you check ? I'm just not sure if i need to put a "break" in the except error or if the return statement do the job ...

**Comment by timgrossmann** [comment]
@texmex44 Seems to work great!

@sionking Can you please double check that?

File: instapy/instapy.py [hunk 1]
```diff
@@ -1316,43 +1324,60 @@ def like_from_image(self, url, amount=50, media=None):
...
-                self.logger.warning(
-                    u'Warning: {} , stopping follow_users'.format(err))
-                return self
-            else:
-                self.logger.error('Sorry, an error occured: {}'.format(err))
-                self.aborting = True
-                return self
+        restart = 0
+        while self.valid_users == 0 or (amount - self.valid_users > 0 and respectAmount):
+            userToInteract = []
```
Comment (Tjorriemorrie) [inline instapy/instapy.py]: use snake case for python, e.g. user_to_interact

File: instapy/instapy.py [hunk 2]
```diff
@@ -1316,43 +1324,60 @@ def like_from_image(self, url, amount=50, media=None):
...
-            if isinstance(err, RuntimeWarning):
-                self.logger.warning(
-                    u'Warning: {} , stopping follow_users'.format(err))
-                return self
-            else:
-                self.logger.error('Sorry, an error occured: {}'.format(err))
-                self.aborting = True
-                return self
+        restart = 0
+        while self.valid_users == 0 or (amount - self.valid_users > 0 and respectAmount):
```
Comment (Tjorriemorrie) [inline instapy/instapy.py]: replace `== 0` with `not self.valid_users`

File: instapy/instapy.py [hunk 3]
```diff
@@ -1316,43 +1324,60 @@ def like_from_image(self, url, amount=50, media=None):
 
         return self
 
-    def interact_user_followers(self, usernames, amount=10, randomize=False):
+    def interact_user_followers(self, usernames, amount=10, randomize=False, respectAmount=False):
```
Comment (Tjorriemorrie) [inline instapy/instapy.py]: respect_amount

File: instapy/instapy.py [hunk 4]
```diff
@@ -1316,43 +1324,60 @@ def like_from_image(self, url, amount=50, media=None):
...
-        self.like_by_users(userToInteract,
-                           self.user_interact_amount,
-                           self.user_interact_random,
-                           self.user_interact_media)
+            self.like_by_users(userToInteract,
+                               self.user_interact_amount,
+                               self.user_interact_random,
+                               self.user_interact_media)
+
+            if respectAmount and self.valid_users > 0 and self.valid_users < amount:
```
Comment (Tjorriemorrie) [inline instapy/instapy.py]: if you make `restart` start with -1 then you don't need the second condition.
If you don't want to change that then you still need to change it to `0 < self.valid_users < amount`. it's also different to the while condition, shouldn't it be the same?

**Review by Tjorriemorrie — CHANGES_REQUESTED** [review]

File: instapy/instapy.py (hunk 1 above)
Comment (texmex44) [reply instapy/instapy.py]: i can change but in the master code `userToInteract` is used.

File: instapy/instapy.py (hunk 2 above)
Comment (texmex44) [reply instapy/instapy.py]: ok

File: instapy/instapy.py (hunk 3 above)
Comment (texmex44) [reply instapy/instapy.py]: ok

File: instapy/instapy.py (hunk 4 above)
Comment (texmex44) [reply instapy/instapy.py]: ok i prefer to keep restart = 0 but i implement your condition `0 < self.valid_users < amount`

**Comment by texmex44** [comment]
@Tjorriemorrie @timgrossmann is it ok ?

**Comment by timgrossmann** [comment]
@texmex44 Sorry for the long delay... 
Could you give a quick explanation in the README what "respect_amount" does? 
We really need to make sure that the documentation stays descriptive and explains all possible features

**Comment by texmex44** [comment]
@timgrossmann sure i will took the time to do this

**Pull request closed without merging**